app/helpers/loss_report_helper.rb

```diff
@@ -0,0 +1,185 @@
+module LossReportHelper
+
+  def loss_report_output(compared_collections, hide_items: true, display: 'text')
+    # depending on the input selection (json or text) a comparison string/hash is created and displayed in-browser
+    # this display feature could be a good candidate for dependency injection
+
+    orig_h = compared_collections[:original_collection_native_hash]
+    conv_h = compared_collections[:translated_collection_native_hash]
+
+    # ISO and DIF collections (in XML form) contain namespaces that cause errors in the below comparison.
+    # Specifically, when nodes are evaluated individually, (their namespace definitions remaining at the top of the xml)
+    # their prefixes are undefined in the scope of the evaluation and therefore raise errors. Removing the namespaces
+    # eliminates this issue.
+    if compared_collections[:native_format].include?('iso') || compared_collections[:native_format].include?('dif')
+      orig = Nokogiri::XML(compared_collections[:original_collection_native_xml]) { |config| config.strict.noblanks }.remove_namespaces!
+      conv = Nokogiri::XML(compared_collections[:translated_collection_native_xml]) { |config| config.strict.noblanks }.remove_namespaces!
+    else
+      orig = Nokogiri::XML(compared_collections[:original_collection_native_xml]) { |config| config.strict.noblanks }
+      conv = Nokogiri::XML(compared_collections[:translated_collection_native_xml]) { |config| config.strict.noblanks }
+    end
+
+    arr_paths = Array.new # This array is used to keep track of the paths that lead to arrays that have already been mapped
+
+    if display == 'text'
+      text_output = String.new
+      text_output += (compared_collections[:native_format] + "\n\n")
```

Comment:
To your question, I meant that in this case you are declaring what `text_output` is, but in subsequent method calls, you are passing both this variable as well as `json_output` which will be nil. So I think it would be helpful to just be explicit and also declare `json_output = nil` here. And vice versa in the elsif below.

Reply:
Ah, I see. will do

Reply:
I missed the added declarations on my first pass. I think that's why we avoid multiple assignment even though ruby allows it

Reply:
Okay just changed it.